test/AbstractAlgebra-test.jl

```diff
@@ -2,19 +2,11 @@ using Random: MersenneTwister
 
 const rng = MersenneTwister()
 
-include("Maps-test.jl")
 include("Groups-test.jl")
 include("NCRings-test.jl")
 include("Rings-test.jl")
 include("Fields-test.jl")
 include("Modules-test.jl")
+include("Maps-test.jl")
 include("Benchmark-test.jl")
 
-@testset "parent_type/elem_type..." begin
```

Comment:
Just as a simple comment: I put this test here because it's not testing something specific to rings, and is implemented in "src/AbstractAlgebra.jl", so having the tests in "test/AbstractAlgebra-test.jl" made sense to me.

Reply:
I've just moved it to the top level where we actually implement a parent_type/elem_type function. No one will be able to find the test otherwise. And it looks ugly in the output.

You are right that it perhaps belongs further up, say at the level of AbstractAlgebra.Set. But as nothing is implemented at that level, I suggest we leave it in Ring. We might even get rid of Set at some point.

But I will now go and remove the implementation from AbstractAlgebra.jl and put it in Rings.jl, because that's not the right place to put implementations.